framework/signature-inference-scripts/infer-and-annotate.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/sh
+
+# This script runs the Checker Framework's signature inference
+# iteratively on a program, adding type annotations on the program, until the
+# .jaif files from one iteration are the same as the .jaif files from the
+# previous iteration.
+
+# This script receives as arguments:
+# 1. Processor's name.
+# 2. Classpath (target project's classpath).
+# 3. Extra processor arguments which will be passed to the checker.
+# 4. List of paths to .java files in a program.
+
+# Example of usage:
+#./infer-and-annotate.sh org.checkerframework.common.value.ValueChecker \
+#   $PROJECT_CP $EXTRA_PROCESSOR_ARGS $PROJECT_DIR/src/my/example/Example1.java \
+#   $PROJECT_DIR/src/my/example/Example2.java
+
+# In case of using this script for Android projects, you must include paths to:
+# android.jar, gen folder, all libs used, source code folder.
+# The Android project must be built with "ant debug" before running this script.
+
+set -e
+
+# This function separates extra arguments passed to the checker from java files
+# received as arguments.
+function read_input() {
+    processor=$1
+    cp=$2:${CHECKERFRAMEWORK}/../annotation-tools/annotation-file-utilities/annotation-file-utilities.jar
+    extra_args=""
+    java_files=""
+    index=2
+    for i in "${@:3}" # Ignores first two arguments (processor and cp).
+    do
+        let "index++"
+        if [[ $i != -* ]]; then
```

Comment:
This makes the assumption that every argument starts with a hyphen.  It means one cannot pass arguments such as -processorpath and -source.  That is at least worth mentioning.